nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -304,4 +377,65 @@ WaypointFollower::goalResponseCallback(
   }
 }
 
+std::vector<geometry_msgs::msg::PoseStamped>
+WaypointFollower::convertGPSPoses2MapPoses(
+  const std::vector<nav2_msgs::msg::OrientedNavSatFix> & gps_poses,
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr & parent_node,
+  const std::unique_ptr<nav2_util::ServiceClient<robot_localization::srv::FromLL>> & fromll_client)
+{
+  RCLCPP_INFO(parent_node->get_logger(), "Converting GPS waypoints to Map Frame..");
+  rclcpp::Duration transform_tolerance(0, 500);
+
+  std::vector<geometry_msgs::msg::PoseStamped> poses_in_map_frame_vector;
+  auto stamp = parent_node->now();
+
+  for (auto && curr_oriented_navsat_fix : gps_poses) {
+    auto request = std::make_shared<robot_localization::srv::FromLL::Request>();
+    auto response = std::make_shared<robot_localization::srv::FromLL::Response>();
+    request->ll_point.latitude = curr_oriented_navsat_fix.position.latitude;
+    request->ll_point.longitude = curr_oriented_navsat_fix.position.longitude;
+    request->ll_point.altitude = curr_oriented_navsat_fix.position.altitude;
+
+    fromll_client->wait_for_service((std::chrono::seconds(1)));
+    if (!fromll_client->invoke(
+        request,
+        response))
+    {
+      RCLCPP_ERROR(
+        parent_node->get_logger(),
+        "fromLL service of robot_localization could not convert %i th GPS waypoint to"
+        "Map frame, going to skip this point!"
+        "Make sure you have run navsat_transform_node of robot_localization");
+      continue;
+    } else {
+      // this poses are assumed to be on global frame (map)
+      geometry_msgs::msg::PoseStamped curr_pose_map_frame;
+      curr_pose_map_frame.header.frame_id = "map";
+      curr_pose_map_frame.header.stamp = stamp;
+      curr_pose_map_frame.pose.position.x = response->map_point.x;
+      curr_pose_map_frame.pose.position.y = response->map_point.y;
+      curr_pose_map_frame.pose.position.z = response->map_point.z;
+
+      geometry_msgs::msg::PoseStamped curr_pose_utm_frame;
+      curr_pose_utm_frame.pose.orientation = curr_oriented_navsat_fix.orientation;
+      curr_pose_utm_frame.header.frame_id = "utm";
+      try {
+        tf_buffer_->transform(
+          curr_pose_utm_frame, curr_pose_map_frame, "map",
+          tf2::durationFromSec(1.0));
```

Comment:
Please use a proper transform tolerance parameter (0.1s should be a good default for that parameter). I see you even make a `transform_tolerance` object but don't use it. Please make a param for that, set declare default to `0.1` and then use that here.